rmw_connext_shared_cpp/src/qos.cpp

```diff
@@ -85,3 +85,120 @@ get_datawriter_qos(
 
   return true;
 }
+
+template<typename DDSEntityQos>
```

Comment:
Could it be that at some point in the future when we support more QoS types, this can no longer be a template?

Reply:
That is why I had to split the functions out - `lifespan` is only present on `DataWriterQos`. However, all the existing generic logic is still valid